pkg/apis/autoscaling/v1/verticalpodautoscalercontroller_types.go

```diff
@@ -17,7 +19,53 @@ type VerticalPodAutoscalerSpec struct {
 	// +kubebuilder:validation:Minimum=0
 	PodMinMemoryMb     *float64 `json:"podMinMemoryMb,omitempty"`
 	RecommendationOnly *bool    `json:"recommendationOnly,omitempty"`
-	MinReplicas        *int64   `json:"minReplicas,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	MinReplicas *int64 `json:"minReplicas,omitempty"`
+	//
+	// +optional
+	DeploymentOverrides DeploymentOverrides `json:"deploymentOverrides"`
+}
+
+// DeploymentOverrides defines overrides for deployments managed by the VerticalPodAutoscalerController
+type DeploymentOverrides struct {
+
+	// admission is the deployment overrides for the VPA's admission container
+	// +optional
+	Admission DeploymentOverride `json:"admission"`
+	// recommender is the deployment overrides for the VPA's recommender container
+	// +optional
+	Recommender DeploymentOverride `json:"recommender"`
+	// updater is the deployment overrides for the VPA's updater container
+	// +optional
+	Updater DeploymentOverride `json:"updater"`
+}
+
+// DeploymentOverride defines fields that can be overridden for a given deployment
+type DeploymentOverride struct {
+	// TODO(jkyros): appsv1.DeploymentSpec fields can go here someday
+
+	// We'd love to just use the deployment spec to override container fields, but it's really
+	// unwieldy having a user fill out the container array and have to merge them by name, so we include
+	// a specific container override field.
+
+	// Container allows for direct overrides on the "important" container of the deployment, i.e. the one
+	// that is actually running the operand
+	// +optional
+	Container ContainerOverride `json:"container"`
+}
```

Comment:
Alright, I left some room for the deployment fields. Right now using it looks like this. Does that feel gross? 
```
apiVersion: v1
items:
- apiVersion: autoscaling.openshift.io/v1
  kind: VerticalPodAutoscalerController
  metadata:
    creationTimestamp: "2023-11-29T18:41:15Z"
    generation: 6
    name: default
    namespace: openshift-vertical-pod-autoscaler
    resourceVersion: "73170"
    uid: 2b544974-2667-42d0-bb50-f8b6d41d291a
  spec:
    deploymentOverrides:
      admission:
        container:
          args:
          - --kube-api-qps=30.0
          - --kube-api-burst=40.0
          resources:
            requests:
              cpu: 40m
              memory: 40Mi
      recommender:
        container:
          args:
          - --kube-api-qps=20.0
          - --kube-api-burst=60.0
          resources:
            requests:
              cpu: 60m
              memory: 60Mi
      updater:
        container:
          args:
          - --kube-api-qps=20.0
          - --kube-api-burst=80.0
          resources:
            requests:
              cpu: 80m
              memory: 80Mi
    minReplicas: 2
    podMinCPUMillicores: 25
    podMinMemoryMb: 250
    recommendationOnly: false
    safetyMarginFraction: 0.15
kind: List
metadata:
  resourceVersion: ""
```